core/src/main/java/org/infinispan/expiration/ExpirationManager.java

```diff
@@ -29,7 +30,17 @@
    void processExpiration();
 
    /**
-    * @return true if expiration is enabled, false otherwise
+    * @return true if expiration reaper is enabled, false otherwise
     */
    boolean isEnabled();
+
+   /**
+    * This should be invoked passing in an entry that is now expired.  Note this method is assumed to not have the
```

Comment:
I think then you need to remove the part about possibly going remote to acquire the lock.